src/Components/Artwork/Save.tsx

```diff
@@ -1,92 +1,154 @@
 import React from "react"
+import * as RelayRuntimeTypes from "relay-runtime"
 import {
   commitMutation,
   createFragmentContainer,
   graphql,
   RelayProp,
 } from "react-relay"
+import { isNull } from "lodash"
 
 import Icon from "../Icon"
-
 import styled from "styled-components"
 import colors from "../../Assets/Colors"
-
 import * as Artsy from "../../Components/Artsy"
 
 const SIZE = 40
 
 export interface Props
   extends RelayProps,
-    React.HTMLProps<SaveButtonContainer>,
-    Artsy.ContextProps {
+    Artsy.ContextProps,
+    React.HTMLProps<React.ComponentType> {
   style?: any
   relay?: RelayProp
+  relayEnvironment?: RelayRuntimeTypes.Environment
   useRelay?: boolean
 }
 
-export class SaveButtonContainer extends React.Component<Props, null> {
-  static defaultProps = {
-    useRelay: true,
-  }
+interface State {
+  is_saved: boolean
+}
+
+export const SaveButtonContainer = Artsy.ContextConsumer(
+  class extends React.Component<Props, State> {
+    static defaultProps = {
+      useRelay: true,
+    }
+
+    state = {
+      is_saved: null,
+    }
+
+    get isSaved() {
+      const isSaved = isNull(this.state.is_saved)
```

Comment:
Just out of JS curiosity, does this do anything else besides `this.state.is_saved == null`

Reply:
[Nope](https://github.com/lodash/lodash/blob/master/isNull.js#L17) -- just reads more at a glance, but not necessary

Reply:
Thanks 👍